content/docs/reference/buildpack-api.md

```diff
@@ -187,13 +186,68 @@ The schema is as follows:
 
         - **`version`** _(string, required)_\
           The version of the buildpack being referred to.
-    
+
         - **`optional`** _(boolean, optional, default: `false`)_\
           Whether or not this buildpack is optional during detection.
- 
+
 - **`metadata`** _(any, optional)_\
     Arbitrary data for buildpack.
- 
+
+## Build Plan
```

Comment:
I have to confess, I read through this section and the linked spec documents a few times and I am still confused :/

Would it be possible to expand the example a bit - e.g., I'm assuming that there are two buildpacks in play here (node 10.x and node 12.x) - perhaps we could walk through the `detect` phase as each buildpack is run and show how the build plan is changing and then what the lifecycle is going to do with it? Or maybe talk about the rationale for why each buildpack specifies the `provides` and `requires` that it does? I think I'm still unsure why a buildpack would both provide and require `node`, for example...

Reply:
I agree. I provided some more specific examples, but I think there still is additional clarifications I could provide. Let me know whether it helped!